docs/debugging-reconfigurator.adoc

```diff
@@ -0,0 +1,1466 @@
+* The current target blueprint is continually being **executed**, meaning that the system is attempting to make the real configuration of the system match the blueprint.
+
+Additionally:
+
+* Blueprints are stored in the control plane database (CockroachDB).
+* The current target blueprint is stored in the control plane database.
+* All (three) Nexus instances are continually running all these processes:
++
+** inventory collection
+** blueprint planning
+** blueprint execution
++
+as well as various other processes involved in managing the system.
+
+All of these are implementation details of the system.  End users are not expected to have to know about, understand, operate, or debug any of this.  Almost none of this is exposed via the external API.
+
+`omdb` (run from the switch zone of either Scrimlet) is the primary tool for support and engineering to observe and control Reconfigurator.
+
+For more, see:
+
+* xref:./reconfigurator.adoc[Reconfigurator documentation]
+* xref:./reconfigurator-dev-guide.adoc[Reconfigurator developer guide]
+* xref:./control-plane-architecture.adoc[Control plane architecture]
+
+== What to expect during a normal software update
+
+=== Customer view
+
+NOTE: See the https://docs.oxide.computer/guides/operator/system-update[end user documentation] for more on the customer view of software update.
+
+Customers start a software update by:
+
+* Using the console, CLI, or API to upload a TUF repository (ZIP file) containing all of new system software; then
+* Setting the system's **target release** to new TUF repository.
+
+The system then asynchronously updates all the software and eventually reports that all software is running the new release.  At that point, from the customer's perspective, the software update is done.
+
+NOTE: When debugging, it's important to understand that internally, the system does not have a notion of a "software update" operation that starts and finishes.  Rather, the system is always and continually trying to make the system's software match whatever the current target release is.  If the operator changes the target release, then the system starts taking steps to update individual components accordingly.  If a component is found to mismatch the target release, even long after the update "completed", the system will take steps to update the component.
+
+The external API provides very basic progress information in the form of the number of compnents running at each software version.  End users (customers) do not have any deeper visibility into what's going on during the update (because communicating more would require customers to understand implementation details of the control plane that they shouldn't need to know to operate the system).
```

Comment:
```suggestion
The external API provides very basic progress information in the form of the number of components running at each software version.  End users (customers) do not have any deeper visibility into what's going on during the update (because communicating more would require customers to understand implementation details of the control plane that they shouldn't need to know to operate the system).
```